ctapipe/reco/HillasReconstructor.py

```diff
@@ -99,13 +99,68 @@ class that reconstructs the direction of an atmospheric shower
 
     """
 
-    def __init__(self, config=None, parent=None, **kwargs):
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
         super().__init__(config=config, parent=parent, **kwargs)
-        self.hillas_planes = {}
-        self.divergent_mode = False
-        self.corrected_angle_dict = {}
+        self.subarray = subarray
+
+        self.cam_R_m = {
+            tel_id: tel.camera.geometry.guess_radius()
+            for tel_id, tel in subarray.tel.items()
+        }
+
+        telframe = TelescopeFrame()
+        self.cam_R_deg = {
+            tel_id: tel.camera.geometry.transform_to(telframe).guess_radius()
+            for tel_id, tel in subarray.tel.items()
+        }
+
+    def __call__(self, event):
+        """
+        Perform the full shower geometry reconstruction on the input event.
+
+        Parameters
+        ----------
+        event : container
+            A `ctapipe` event container
+        """
+
+        # Read only valid HillasContainers
+        hillas_dict = {
+            tel_id: dl1.parameters.hillas
+            for tel_id, dl1 in event.dl1.tel.items()
+            if np.isfinite(event.dl1.tel[tel_id].parameters.hillas.intensity)
+        }
+
+        # This is here because technically, due to tracking, it is not a
+        # the pointing of the array is not a constant
+        array_pointing = SkyCoord(
+            az=event.pointing.array_azimuth,
+            alt=event.pointing.array_altitude,
+            frame=AltAz(),
+        )
+
+        telescope_pointings = {}
```

Comment:
unneeded line (directly overriden in next)

Reply:
removed